tests/e2e/main.py

```diff
@@ -0,0 +1,124 @@
+            )
+            print("Generated log event: %s" % log_event_retry)
+        else:
+            raise(e)
+
+def search_dynatrace_log_records(url_prefix,
+                                 message_content):
+    """
+    (using Dynatrace API) return a list of log records matching the specified query
+    """
+    url = url_prefix + '/logs/search?' + 'from=now-5m&limit=1000&query=aws.service%3D%22lambda%22%20AND%20content%3D%22' + message_content + '%22%20AND%20timestamp%3D%22' + str(epoch_timestamp_in_ms) + '%22&sort=-timestamp'
+    response = requests.get(url,
+                            headers=request_headers)
+
+    # check, if response status code is OK
+    if response.status_code != requests.codes.ok:
+        response.raise_for_status()
+
+    return response.json()
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--url-prefix',
+                        '-u',
+                        action='store',
+                        type=str,
+                        default='https://jxw01498.dev.dynatracelabs.com/api/v2',
+                        help='specify Dynatrace SaaS URL or Dynatrace AG URL')
+    parser.add_argument('--api-token',
+                        action='store',
+                        type=str,
+                        default='',
+                        required=True,
+                        help='specify Dynatrace API Token')
+    parser.add_argument('--log-group-name',
+                        '-l',
+                        action='store',
+                        type=str,
+                        default='',
+                        required=True,
+                        help='specify AWS CloudWatch Logs log group name')
+    parser.add_argument('--log-stream-name',
+                        '-s',
+                        action='store',
+                        type=str,
+                        default='',
+                        required=True,
+                        help='specify AWS CloudWatch Logs log stream name for a given log group')
+    parser.add_argument('--unique-message-content',
+                        '-m',
+                        action='store',
+                        type=str,
+                        default='Dynatrace-AWS-Log-Forwarder-End-to-End-test-message',
+                        help='specify a unique log event message content')
+    args = parser.parse_args()
+
+    epoch_timestamp_in_ms = int(time.time() * 1000)
+    request_headers       = {'Authorization': 'Api-Token ' + args.api_token,
+                            'Content-Type': 'application/json',
+                            'charset': 'utf-8'}
+    message_content       = 'Dynatrace-AWS-Log-Forwarder-End-to-End-test-' + args.unique_message_content
+
+    # Generate a test log event
+    generate_log_event(args.log_group_name,
+                       args.log_stream_name)
+
+    # Search Dynatrace logs for records with specific message content and timestamp
```

Comment:
instead of writing your own method to test use pytest as we have in 2 other projects. pytest is used also for unit and integration

Reply:
We had a discussion concerning the above during today's daily and decided to stick with the current approach. In our case re-writing to `pytest`, considering a more complicated script and my lack of knowledge on the subject of using the lib for scenarios other, than unit testing, would require more effort.

We could create a task for a future re-factor/improvement.